.github/workflows/test-upstream.yaml

```diff
@@ -111,43 +111,40 @@ jobs:
           git remote add upstream https://github.com/linkml/linkml-runtime
           git fetch upstream --tags
 
-      - name: set up python
-        uses: actions/setup-python@v5.6.0
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
         with:
+          enable-cache: true
+          version: ${{ env.UV_VERSION }}
           python-version: ${{ matrix.python-version }}
 
-      - name: Install poetry
-        run: |
-          pipx install poetry
-          pipx inject poetry poetry-dynamic-versioning
-
       - name: Load cached venv
-        id: cached-poetry-dependencies
+        id: cached-uv-dependencies
         uses: actions/cache@v4.2.3
         with:
           path: linkml/.venv
-          key: venv-${{ matrix.python-version }}-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+          key: venv-${{ matrix.python-version }}-${{ runner.os }}-${{ hashFiles('**/uv.lock') }}
```

Comment:
You can remove this cache section as in https://github.com/linkml/linkml/pull/2779/files#diff-64b6ec5d5ea1cf152a537d4c35d84522a472ee91d1b661a40017c0cb1e71089b

I have also seen that this caching caused flaky job failures for Windows tests.